packages/web3-eth-accounts/src/index.js

```diff
@@ -320,6 +320,24 @@ function Wallet(accounts) {
     this.defaultKeyName = "web3js_wallet";
 }
 
+Wallet.prototype._findSafeIndex = function (pointer) {
+    pointer = pointer || 0;
+    if (_.has(this, pointer)) {
+        return this._findSafeIndex(pointer + 1);
+    } else {
+        return pointer;
+    }
+};
+
+Wallet.prototype._currentIndexes = function () {
+    var keys = Object.keys(this);
+    var indexes = keys
+        .map(function(key) { return parseInt(key); })
+        .filter(function(n) { return (n < 9e20); });
```

Comment:
Filters out NaN's and hex addresses (which will be parsed to int's). Makes the assumption that a user will not have many quintillions of wallets.